src/eth/relayer/external.rs

```diff
@@ -562,6 +557,21 @@ impl ExternalRelayer {
             };
         }
 
+        mismatched_blocks
+    }
+
+    // Split the dag and relay its components.
+    #[tracing::instrument(name = "external_relayer::relay_dag", skip_all)]
+    async fn relay_dag(&self, dag: TransactionDag) -> MismatchedBlocks {
+        let start = Instant::now();
+
+        tracing::debug!("relaying transactions");
+
+        let futures = dag.split_components().into_iter().map(|dag| self.relay_component(dag));
+        let results = join_all(futures).await;
```

Comment:
It shooooouldn't be a problem, but I'd use `buffered` on this one too instead of the `join_all` call, probably with a large buffer size tho.

Reply:
True, will improve this in the next PR.